x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -100,18 +100,48 @@ export const hasReadIndexPrivileges = async (
   return false;
 };
 
+const getFieldCapFailingIndices = (
+  inputIndices: string[], // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  timestampFieldCapsResponse: ApiResponse<Record<string, any>, Context>,
+  timestampField: string
+) => {
+  if (
+    isEmpty(timestampFieldCapsResponse.body.fields) &&
+    isEmpty(timestampFieldCapsResponse.body.indices)
+  ) {
+    return inputIndices;
+  } else if (
+    isEmpty(timestampFieldCapsResponse.body.fields) &&
+    !isEmpty(timestampFieldCapsResponse.body.indices)
+  ) {
+    return timestampFieldCapsResponse.body.indices;
+  } else {
+    return timestampFieldCapsResponse.body.fields[timestampField].unmapped.indices;
+  }
+};
+
 export const hasTimestampFields = async (
   wroteStatus: boolean,
   timestampField: string,
   // any is derived from here
   // node_modules/@elastic/elasticsearch/api/kibana.d.ts
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   timestampFieldCapsResponse: ApiResponse<Record<string, any>, Context>,
+  inputIndices: string[],
   ruleStatusService: RuleStatusService,
   logger: Logger,
   buildRuleMessage: BuildRuleMessage
 ): Promise<boolean> => {
-  if (
+  if (!wroteStatus && isEmpty(timestampFieldCapsResponse.body.indices)) {
+    const errorString = `The following indices are missing the ${
```

Comment:
`The following index patterns did not match any indices: ${JSON.stringify(inputIndices)}` might be a clearer error message.